### PR TITLE
BUG: propagate dropna in pd.Grouper

### LINKED ISSUE #36620: BUG: Grouper should have a dropna attribute
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

xref #35612

#### Code Sample, a copy-pastable example

```python
In [1]: import pandas as pd 
   ...: df = pd.DataFrame({"A": [0, 0, 1, None], "B": [10, 2, 10, None]}) 
   ...: gb = df.groupby('A', dropna=False)                                     

In [2]: gb.dropna                                                              
Out[2]: False

In [3]: gb.grouper.dropna                                                      
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-4d8081f9c78e> in <module>
----> 1 gb.grouper.dropna

AttributeError: 'BaseGrouper' object has no attribute 'dropna'
```

#### Problem description

Now that we support a `dropna` argument to `DataFrameGroupBy` and `SeriesGroupBy` we need to be able to use it in `Grouper` (#35751)

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : f34a56b4ddcf51a8a03a9d20f07ab039f64ff2e6
python           : 3.7.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-48-generic
Version          : #52-Ubuntu SMP Thu Sep 10 10:58:49 UTC 2020
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : C.UTF-8
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.2.0.dev0+484.gf34a56b4d
numpy            : 1.18.5
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 45.2.0.post20200210
Cython           : 0.29.21
pytest           : 5.4.3
hypothesis       : 5.19.0
sphinx           : 3.1.1
blosc            : None
feather          : None
xlsxwriter       : 1.2.9
lxml.etree       : 4.4.1
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.16.1
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : 1.3.2
fsspec           : 0.7.4
fastparquet      : 0.4.0
gcsfs            : 0.6.2
matplotlib       : 3.2.1
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.4
pandas_gbq       : None
pyarrow          : 0.16.0
pytables         : None
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.5.0
sqlalchemy       : 1.3.18
tables           : 3.6.1
tabulate         : 0.8.7
xarray           : 0.15.1
xlrd             : 1.2.0
xlwt             : 1.3.0
numba            : 0.50.1

</details>

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes #36620
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

A precursor to #35751

### CODE REVIEW

File: pandas/tests/groupby/test_groupby_dropna.py [hunk 1]
```diff
@@ -162,6 +162,13 @@ def test_groupby_dropna_series_by(dropna, expected):
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize("dropna", (False, True))
+def test_Grouper_dropna_propagation(dropna):
+    df = pd.DataFrame({"A": [0, 0, 1, None], "B": [1, 2, 3, None]})
+    gb = df.groupby("A", dropna=dropna)
+    assert gb.grouper.dropna is dropna
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_groupby_dropna.py:169]: I think use == instead of is here.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
lgtm, one small request.

File: pandas/tests/groupby/test_groupby_dropna.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/tests/groupby/test_groupby_dropna.py:169]: Also, lowercase g

Comment (arw2019) [reply pandas/tests/groupby/test_groupby_dropna.py]: Thanks for looking this over! Fixed both of these

**Comment by arw2019** [comment]
I noticed that the `Grouper` docstring didn't have a `dropna` arg so I added that here.

I said `added in 1.2.0` - a bit of a slight of hand since it the attribute was already there in the code in `Grouper` (but it didn't exist/propagate in `BaseGrouper` so it was unusable)

File: pandas/tests/groupby/test_groupby_dropna.py [hunk 2]
```diff
@@ -162,6 +162,13 @@ def test_groupby_dropna_series_by(dropna, expected):
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize("dropna", (False, True))
+def test_grouper_dropna_propagation(dropna):
+    df = pd.DataFrame({"A": [0, 0, 1, None], "B": [1, 2, 3, None]})
+    gb = df.groupby("A", dropna=dropna)
+    assert gb.grouper.dropna == dropna
```
Comment (WillAyd) [inline pandas/tests/groupby/test_groupby_dropna.py:169]: Does this have any user facing impact? The Grouper itself is not a public object

Comment (arw2019) [reply pandas/tests/groupby/test_groupby_dropna.py:169]: > Does this have any user facing impact? The Grouper itself is not a public object

Doubt it. I only added this while working on groupby(dropna=...) stuff.

`Grouper` does have a page in the api ref though so it's at least nominally exposed:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Grouper.html

File: doc/source/whatsnew/v1.2.0.rst [hunk 3]
```diff
@@ -349,6 +349,7 @@ Groupby/resample/rolling
 - Bug in :meth:`DataFrame.groupby` does not always maintain column index name for ``any``, ``all``, ``bfill``, ``ffill``, ``shift`` (:issue:`29764`)
 - Bug in :meth:`DataFrameGroupBy.apply` raising error with ``np.nan`` group(s) when ``dropna=False`` (:issue:`35889`)
 - Bug in :meth:`Rolling.sum()` returned wrong values when dtypes where mixed between float and integer and axis was equal to one (:issue:`20649`, :issue:`35596`)
+- Bug in :class:`pd.Grouper` now propagates ``dropna`` argument correctly (:issue:`36620`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:352]: i would remove this for now; in the followup which actually changes behavior you can add what is changed. this doesn't do anything per-se.

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.2.0.rst (hunk 3 above)
Comment (arw2019) [reply doc/source/whatsnew/v1.2.0.rst]: Done

File: pandas/tests/groupby/test_groupby_dropna.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/tests/groupby/test_groupby_dropna.py:170]: @WillAyd maybe you're thinking we should only be testing the public API and not any internals?

**Comment by WillAyd** [comment]
The test is fine it was a general question in the PR. Was addressed by Jeff as well so otherwise lgtm

**Review by rhshadrach — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ok this is fine

File: pandas/core/groupby/ops.py [hunk 4]
```diff
@@ -97,6 +98,7 @@ def __init__(
         self.group_keys = group_keys
         self.mutated = mutated
         self.indexer = indexer
+        self.dropna = dropna
```
Comment (jbrockmendel) [inline pandas/core/groupby/ops.py:101]: @arw2019 this attribute isnt used anywhere except for in the test added in this PR.  is it still needed?  is it part of a precursor to something on the horizon?

**Pull request merged**